7 - Allow models to pull from local

Jared-Krajewski opened the pull request:
resolves #7. Added `make copy-model` so you can pull models from local instead of having to re-download them. If there are any errors with the copy model, it will just pull the model like it previously was.  this is all default with `make build`

I also pulled the remove volume from make clean so downloaded models will persist between builds either way. Make clobber can be used to clear the volumes or I can add a separate make command to also clear the volumes 
___

Update:  resource sharing is no longer necessary after latest push. removing images and writeup in regards to that

Jared-Krajewski committed:
added make copy-model with a fallback to pull model, renamed utility folder to tools, updated docker volume name for ollama

leekahung commented:
Is there a way to do this without configuring docker desktop? Like a script or cli command?

Jared-Krajewski commented:
> Is there a way to do this without configuring docker desktop? Like a script or cli command?

can do a sudo write to the file that docker updates, but it seemed to be really frowned upon. I'll do a little reading and see if I can find another way.

also just noticed I updated some make commands but not in the build, so ill fix that.

Jared-Krajewski committed:
updated clone copy-model to not require host sharing, makefile build updates after command renaming

Jared-Krajewski commented:
nvm, I was just over-complicating this from the start [docker](https://docs.docker.com/get-started/docker-concepts/running-containers/sharing-local-files/) had it pretty straight forward. 

updated that and fixed the make commands. I removed the shared resource from my docker and tested it and it worked.

leekahung reviewed (approved):
Looks good 👍
